server/ethereum/ethereum.test.js

```diff
@@ -29,18 +30,87 @@ afterAll(() => {
   web3.currentProvider.connection.close()
 })
 
-// NOTE - this isn't really testing anything yet, just showing how to fire an event and getting the testing framework working
-test('firing an event pins a file', async done => {
+test('watcher pins file from registerData function', async done => {
   const testKey = web3.utils.fromAscii('testKey')
-  const testVal = { dag: 'killinnnnnn it' }
-  const hash = await node.dag.put(testVal)
+  const hash = await node.dag.put(dag)
 
   registerWatcher(contract)
   contract.methods
     .registerData(testKey, hash.toBaseEncodedString())
     .send({ from: accounts[0] }, () => {
-      setTimeout(() => {
+      setTimeout(async () => {
+        const pins = await node.pin.ls()
+        const match = pins.find(item => {
+          return item.hash === hash.toBaseEncodedString()
+        })
+        expect(match).toBeDefined()
         done()
       }, 2000)
     })
 })
+
+test('handlePinHashEvent pins file of cid it was passed', async done => {
+  dag.testKey = 'handlePinHashEventTestVal'
+  const hash = await node.dag.put(dag)
+  const eventObj = {
+    returnValues: {
+      cid: hash.toBaseEncodedString()
+    }
+  }
+
+  const res = await handlePinHashEvent(null, eventObj)
+  expect(res[0].hash).toBe(hash.toBaseEncodedString())
+  done()
+})
+
+test('handlePinHashEvent throws error with empty params', async done => {
+  try {
+    await handlePinHashEvent()
+    expect(true).toBe(false)
+    done()
+  } catch (error) {
+    expect(error).toBeDefined()
+    done()
+  }
+})
+
+test('getContract returns a contract', async done => {
+  const contract = getContract(
+    demoSmartContractJson1,
+    demoSmartContractJson1.address
+  )
+  expect(contract._address).toBe(demoSmartContractJson1.address)
+  done()
+})
+
+test('getContract throws when an invalid contract is passed', async done => {
+  demoSmartContractJson1.address = '0x7505462c30102eBCDA555446c3807362AeFEfc8r'
+  const badCall = () => {
+    return getContract(
+      demoSmartContractJson1,
+      '0x7505462c30102eBCDA555446c3807362AeFEfc8r'
+    )
+  }
+
+  expect(badCall).toThrow()
+  done()
+})
+
+// Uncomment this when timeout in ipfs.dag.get is working. Will need to adjust jest.setTimeout.Timeout past the dag.get timeout time.
```

Comment:
should we wait until that time to merge? I thought we agreed no more incomplete tests in primary unless someone is blocked without your code

Reply:
my assumption was that invalid Cid shouldn't be timeout error - ipfs should just throw when you call it with bad Cid. is this wrong?

Reply:
It's a complete test, but there is an error on the ipfs side. It doesn't timeout.

Reply:
It's just ensuring it times out.

Reply:
I like reminders like this myself.

Reply:
But we can remove if y'all don't want these. I can make a ticket or something instead.

Reply:
I think a ticket is better/cleaner but that's just personal style bc I don't look for reminders in code

Reply:
but if its a malformed CID shouldn't it just throw immediately, not timeout?

Reply:
Actually, to make the case for comments, this is a really common pattern. I.e. someone pushes a bunch of code that needs to go in but we don't have time to fix them. Or something out of our control breaks and we need to wait until it's fixed. If we have to create tickets that creates way more overhead I think. And it's not that we're looking for reminders, but as long as they're there in comments we can pick them up later when we're back in test mode.

Reply:
And ya, it'll throw right away if it's malformed. But it's not malformed, but not found on the network, we want to make sure it times out.

Reply:
resolving this convo for a suggestion below